modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -241,20 +296,20 @@ interface PageUrlHandler {
  *
  * @returns {string} The generated URL.
  */
-export function pageUrl(params: PageUrlParams): string {
+export function pageUrl<Params extends RequestParams = RequestParams>(params: PageUrlParams<Params>): string {
```

Comment:
For all of these functions (`pageUrl`, `serviceUrl`) both `number` and `boolean` is valid types of the value (and `object` are just url-encoded, but that might be more coincidental).

Should maybe `Params` not extend `RequestParams` but another shape instead that allows for other types of the values.